packages/medusa/src/strategies/price-selection.ts

```diff
@@ -26,12 +31,122 @@ class PriceSelectionStrategy extends AbstractPriceSelectionStrategy {
     return new PriceSelectionStrategy({
       manager: manager,
       moneyAmountRepository: this.moneyAmountRepository_,
+      featureFlagRouter: this.featureFlagRouter_,
     })
   }
 
   async calculateVariantPrice(
     variant_id: string,
     context: PriceSelectionContext
+  ): Promise<PriceSelectionResult> {
+    if (
+      this.featureFlagRouter_.isFeatureEnabled(
+        TaxInclusivePricingFeatureFlag.key
+      )
+    ) {
+      return this.calculateVariantPrice_new(variant_id, context)
+    }
+    return this.calculateVariantPrice_old(variant_id, context)
```

Comment:
praise: think this is a very nice approach 👍

Reply:
thanks, I thought about the tweet you mentioned 😄 do you think we should add a check when the feature flag is enabled to validate the result (if the prices are exclusive): 
```
const pricing = this.calculateVariantPrice_new(variant_id, context)

if(env.node_env === 'development'){
  const oldPricing = this.calculateVariantPrice_old(variant_id, context)

  const err = []
  if(!pricing.calculated_includes_tax && oldPricing.calculated_price !== pricing.calculated_price) {
    err.push('calculated price from the old price calculation doesn't match the price from the new price calculation')
  }

  if(!pricing.original_includes_tax && oldPricing.original_price !== pricing.original_price) {
    err.push('original price from the old price calculation doesn't match the price from the new price calculation')
  }

  if(err.length){
    throw new MedusaError(MedusaError.Types.INVALID_DATA, err.join(", "))
  }
}
return pricing
```

Reply:
I think we are okay without it :)